### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by alvaroiporre**
## Project Requirements

In this PR I made the following:

- Created the Nameable interface in the nameable.rb file.
- Created the BaseDecorator class in the base_decorator.rb file.
- Created the CapitalizeDecorator class in the capitalize_decorator.rb file.
- Created the TrimmerDecorator class in the trimmer_decorator.rb file.
- Tested the above classes in the main.rb file.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require_relative 'base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    return @nameable.correct_name.strip if @nameable.correct_name.length > 10
+
+    @nameable.correct_name
+  end
```
Comment (danmainah) [inline trimmer_decorator.rb]: - [ ] @alvaroiporre Great work on your decoration, however, the TrimmerDecorator class doesn't work, below is the result from your main.rb file testing the decoration you can see the name doesn't get trimmed to only 10 characters
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @alvaroiporre , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ added all classes for decoration following the [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
✔️ Professional written README file with all sections required.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: trimmer_decorator.rb (hunk 1 above)
Comment (alvaroiporre) [reply trimmer_decorator.rb]: Thank you so much for your feedback @danmainah. I can't understand how this decorator must be work. Can you please give me an example to fix the trimmerDecorator.

**Comment by alvaroiporre** [comment]
I already fixed the required changes.

**Review by Gambit142 — APPROVED** [review]
Hi @alvaroiporre 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linters checks passed ✔️ 
- Created `Nameable` class ✔️ 
- Created a Base Decorator class ✔️ 
- Created other decorators ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**